app/src/api/api.ts

```diff
@@ -93,14 +95,30 @@ server.get('/search', function (req, res) {
     });
 });
 
-server.use((err, req, res, next) => {
+server.use((err: any, req: express.Request, res: express.Response, next: express.NextFunction) => {
     if (res.headersSent) {
         return next(err);
     }
 
     if (err != undefined) {
         console.log('Global error handler: ', err);
-        res.status(500).send({ error: 'Server error' });
+        
+        if (err instanceof ApiUserError) {
+            let errorMessage: string;
+
+            if(err instanceof ApiParamError) {
+                errorMessage = `Problem with parameter ${err.paramName}: ${err.message}`;
+            } else {
+                errorMessage = err.message;
+            }
+
+            res.status(400).send({ error: errorMessage });
+        } else if(err instanceof DatabaseError){
+            res.status(500).send({ error: 'Database error' });
+        } else {
+            res.status(500).send({ error: 'Server error' });
+        }
+        
```

Comment:
Add global handling for typical error situations - this way the individual controllers don't need to all include the code to handle database errors, or errors related to parsing a HTTP request parameter.